pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -38,6 +39,7 @@ func (p *Pipeline) Validate(ctx context.Context) *apis.FieldError {
 	if apis.IsInDelete(ctx) {
 		return nil
 	}
+	ctx = config.SetValidateParameterVariablesAndWorkspaces(ctx, true)
```

Comment:
I should have mentioned this on the other PR, but this name is confusing to me. Just based on the name alone, it's not really clear what changes when you set this. It feels like it's saying "ok, validate parameters and workspaces" in general, which seems like something you'd never actually want to turn off. Could we get this a more clear name to better signify exactly what it's doing?

Reply:
Yes, that's right. How about I have a flag that is defaulted to `true` and we only call it when we intend to make it `false`? 

e.g.

we could have `config.DoNotValidateParameterVariablesAndWorkspaces(ctx)` which would set it the underlying context field to `false`. 

We also have a function `config.ValidateParameterVariablesAndWorkspaces(ctx)` which returns a `true` or `false` based on the value of the context field. This function is called by `task_validation` to decide whether to validate parameters and workspaces. We can have it return `true` by default.

Does that look like a good solution?

Reply:
I'd really like the name to be more clear than `ValidateParameterVariablesAndWorkspaces` - that name is just too vague. What does it relate to? Which parameter variables and workspaces? If we're just using it for parameter propagation cases, it feels like it would make more sense to have the name be something regarding parameter propagation validation, i.e., _why_ we're setting this flag, not what the result of setting this flag is.

Reply:
I see what you mean. Its trying to skip validation when it does not contain the full information. e.g. When propagating parameters, a user could provide the params at the task run or the pipeline run level and directly call them in the steps. However, when we validate the underlying taskSpec or pipelineSpec, we don't have these params declared there and so the web hook would throw an error. This flag allows us to specify where to skip validation of params due to lack of information.

I could call the it `SkipValidationDueToPropagatedParametersAndWorkspaces` (a bit of a mouthful, but ok I guess)?

Reply:
That would definitely be better, yeah.

Reply:
I will make the appropriate changes.

Reply:
@abayer done! Thanks for the suggestion.